product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/utils/XMLComparatorInterop.java

```diff
@@ -0,0 +1,212 @@
+
+        log.debug("Comparing Children ......");
+        status = compareAllChildren(elementOne, elementTwo);
+
+        return status;
+    }
+
+    private static boolean compareAllAttributes(OMElement elementOne,
+                                                OMElement elementTwo) {
+        boolean status;
+        status = compareAttributes(elementOne, elementTwo);
+        status = compareAttributes(elementTwo, elementOne);
+        return status;
+    }
+
+    private static boolean compareAllChildren(OMElement elementOne,
+                                              OMElement elementTwo) {
+        boolean status;
+        status = compareChildren(elementOne, elementTwo);
+        return status;
+    }
+
+
+    private static boolean isIgnorable(OMElement elt) {
+        if (elt != null) {
+            OMNamespace namespace = elt.getNamespace();
+            if (namespace != null) {
+                return ignorableNamespaceList.contains(namespace.getNamespaceURI());
+            } else {
+                return false;
+            }
+        } else {
+            return false;
+        }
+    }
+
+
+    private static boolean compareChildren(OMElement elementOne, OMElement elementTwo) {
+        //ignore if the elements belong to any of the ignorable namespaces list
+        boolean status = true;
+        if (isIgnorable(elementOne) ||
+                isIgnorable(elementTwo)) {
+            return true;
+        }
+        Iterator elementOneChildren = elementOne.getChildren();
+        while (elementOneChildren.hasNext()) {
+            OMNode omNode = (OMNode) elementOneChildren.next();
+            if (omNode instanceof OMElement) {
+                OMElement elementOneChild = (OMElement) omNode;
+                OMElement elementTwoChild = null;
+                //Do the comparison only if the element is not ignorable
+                if (!isIgnorable(elementOneChild)) {
+                    Iterator elementTwoChildren = elementTwo.getChildren();
+                    while (elementTwoChildren.hasNext()) {
+                        status = false;
+                        OMNode node = (OMNode) elementTwoChildren.next();
+                        if (node.getType() == OMNode.ELEMENT_NODE) {
+                            elementTwoChild = (OMElement) node;
+                            if (elementTwoChild.getLocalName()
+                                    .equals(elementOneChild.getLocalName())) {
+                                //Do the comparison only if the element is not ignorable
+                                if (!isIgnorable(elementTwoChild)) {
+                                    if (elementTwoChild == null) {
+                                        return false;
+                                    }
+                                }
+
+                                status = compare(elementOneChild, elementTwoChild);
+
+                            }
+                        }
+                        if (status) {
```

Comment:
if (status) does not have a clear meaning. Shall we rename the variable with a meaningful one?